demos/demo.cpp

```diff
@@ -0,0 +1,171 @@
+/**
+ * @file demo.cpp
+ * @author Vincent Berenz
+ * license License BSD-3-Clause
+ * @copyright Copyright (c) 2019, Max Planck Gesellschaft.
+ * 
+ * @brief Minimal demo of robot driver, backend and frontend
+ */
+
+#include "robot_interfaces/robot_driver.hpp"
+#include "robot_interfaces/robot_backend.hpp"
+#include "robot_interfaces/robot_frontend.hpp"
+
+#include <memory>
+
+/**
+ * \example demo.cpp
+ * This demo shows robot_interfaces of a 
+ * dummy "2dof" robot, in which a dof "position"
+ * is represented by an integer
+ */
+
+
+
+// Actions to be performed by robot, will be received by Driver
+// An action simply encapsulate two desired position value,
+// one for each dof
+class Action
+{
+  
+public:
+  
+  int values[2];
+
+  void print(bool backline)
+  {
+    std::cout << "action: "
```

Comment:
i think the spacing here does not follow the conventions (4 spaces)

Reply:
@mwuethri during the upcoming cleaning week, I plan to update all the code I developed to the convention.